pkg/api/validation/validation.go

```diff
@@ -1242,6 +1242,29 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldForbidden("data", "Maximum secret size exceeded"))
 	}
 
+	switch secret.Type {
+	case api.SecretTypeOpaque, "":
+		// no-op
+	default:
+		allErrs = append(allErrs, errs.NewFieldInvalid("type", secret.Type, "unknown SecretType"))
```

Comment:
@liggitt I think we should take this validation out for now and open an issue re: types and whether they can be user defined.